crmsh/bootstrap.py

```diff
@@ -104,6 +105,179 @@ def init_qdevice(self):
+            invoke("systemctl enable sbd.service")
+        else:
+            invoke("systemctl disable sbd.service")
+
+    def configure_sbd_resource(self):
+        if self.sbd_device and self.configured_sbd_device():
+            if not invoke("crm configure primitive stonith-sbd stonith:external/sbd pcmk_delay_max=30s"):
+                error("Can't create stonith-sbd primitive")
+            if not invoke("crm configure property stonith-enabled=true"):
+                error("Can't enable STONITH for SBD")
+        elif self.diskless_sbd:
+            if not invoke("crm configure property stonith-enabled=true stonith-watchdog-timeout=5s"):
+                error("Can't enable STONITH for diskless SBD")
+
+    def _get_sbd_device(self):
+        if self.sbd_device:
+            self.sbd_device = self._parse_sbd_device()
+            self._verify_sbd_device()
+        elif not self.diskless_sbd:
+            self._get_sbd_device_interactive()
+
+    def _initialize_sbd(self):
+        if self.diskless_sbd:
+            return
+        status_long('Initializing SBD...')
+        for dev in self.sbd_device:
+            if not invoke("sbd -d {} create".format(dev)):
+                error("Failed to initialize SBD device {}".format(dev))
+        status_done()
+
+    def _update_configuration(self):
+        self.sbd_config_dict["SBD_WATCHDOG_DEV"] = detect_watchdog_device()
+        if self.sbd_device:
+            self.sbd_config_dict["SBD_DEVICE"] = ';'.join(self.sbd_device)
+        utils.sysconfig_set(SYSCONFIG_SBD, **self.sbd_config_dict)
+        csync2_update(SYSCONFIG_SBD)
+
+    def _get_sbd_device_interactive(self):
+        if _context.yes_to_all:
+            warn("Not configuring SBD (%s left untouched)." % (SYSCONFIG_SBD))
+            return
+
+        status("""
+Configure SBD:
+  If you have shared storage, for example a SAN or iSCSI target,
+  you can use it avoid split-brain scenarios by configuring SBD.
+  This requires a 1 MB partition, accessible to all nodes in the
+  cluster.  The device path must be persistent and consistent
+  across all nodes in the cluster, so /dev/disk/by-id/* devices
+  are a good choice.  Note that all data on the partition you
+  specify here will be destroyed.
+""")
+
+        if not confirm("Do you wish to use SBD?"):
+            warn("Not configuring SBD - STONITH will be disabled.")
+            return
+
+        configured_dev = self.configured_sbd_device()
+        if configured_dev and not confirm("SBD is already configured to use {} - overwrite?".format(configured_dev)):
+            return
+
+        dev_looks_sane = False
+        while not dev_looks_sane:
+            dev = prompt_for_string('Path to storage device (e.g. /dev/disk/by-id/...), or "none", use ";" as separator for multi path', r'none|\/.*')
+            if dev == "none":
+                self.diskless_sbd = True
+                return
+            dev_list = dev.strip(';').split(';')
```

Comment:
I think we can somehow use `_verify_sbd_device` method here, the code is exactly the same. Not just copy/pasting, what with some minor changes it might be done

Reply:
Changed, reuse `_verify_sbd_device`